sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridPublisherClient.cs

```diff
@@ -122,16 +122,18 @@ private async Task<Response> SendEventsInternal(IEnumerable<EventGridEvent> even
                     JsonDocument data;
                     if (egEvent.Data is BinaryData binaryEventData)
                     {
-                        data = JsonDocument.Parse(binaryEventData);
+                        try
+                        {
+                            data = JsonDocument.Parse(binaryEventData);
```

Comment:
This somewhat highlights where a DataFormat property on BinaryData might help. We don't know whether the BinaryData is already serialized or not, so we have to do this try/catch.